### PR TITLE
Fix cmake for usage of the slider as a library and add an editEnded signal

### PR DESCRIPTION
**Pull request by otto-link**
Hey, I'm revamping the UI of a node-based application I'm developping (for procedural terrain generation -> https://github.com/otto-link/Hesiod). I'm planning to use your widgets because they are really good looking and the bound check is really interesting!

I added some minor modifications:
efdde64a72290bbe5bb070f3a0ccb4cad51ae579 : I had to modify the cmake file in order to make it work out of the box for my needs.
02deaf9e11a9395717a814093099fd15a620632d : I added a new signal, triggered only when the editing is finished. For my case, to spare some computational resources, I just want to use the final value chosen by the user.

I hope you'll consider this as PR material. Let me know if you have any questions.

### CODE REVIEW

**Review by BugelNiels — CHANGES_REQUESTED** [review]
Thanks for the contribution! Apart from some indentation issues the changes look good.
To make the review process a bit smoother from my side in the future, I've set up some basic CI and unit tests. If you rebase your branch on top of the current `main` and fix the indentation issues, then everything should be good.

Note that I  had to add `set(CMAKE_AUTOMOC ON)` for the tests to work, so this change has already been integrated.

File: src/valueslider.cpp [hunk 1]
```diff
@@ -181,7 +181,8 @@ void ValueSliders::ValueSlider<T>::mouseReleaseEvent(QMouseEvent *event) {
         if (event->button() == Qt::LeftButton) {
             QCursor::setPos(startPos_);
             updateValueByPosition(event->pos().x() - oldPos_);
-        }
+	    emitEditEnded();
+	}
```
[the remark this answers is not in the document]
Comment (otto-link) [reply src/valueslider.cpp]: ok, replaced tabs with spaces

File: src/valueslider.cpp [hunk 2]
```diff
@@ -236,6 +237,7 @@ void ValueSliders::ValueSlider<T>::submitTypedInput() {
     T newVal = convertString(typeInput_, ok);
     if (ok) {
         setVal(newVal);
+	emitEditEnded();
```
[the remark this answers is not in the document]
Comment (otto-link) [reply src/valueslider.cpp]: ok, replaced tabs with spaces

**Comment by otto-link** [comment]
Hi, thanks for the review and the time spent on it. I took into account your feedback, the PR should be good. Let me know if there are still some modifications to make.

**Comment by BugelNiels** [comment]
image
It seems that the rebase was not done properly. Your changes should be on top of those in the main branch, as shown by the commits in the screenshot above, there are some changes before and some changes after. That is also why the `Files changed` part shows a bunch of files that weren't edited. Please try to fix it so that this PR only shows the changes you made so that the git history stays clean

**Comment by otto-link** [comment]
Indeed, sorry for the mess, I resubmitted a PR to ease things out...

**Pull request closed without merging**